test_unstructured_ingest/src/local-embed-vertexai.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+set -e
+
+SRC_PATH=$(dirname "$(realpath "$0")")
+SCRIPT_DIR=$(dirname "$SRC_PATH")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=embed-vertexai
+OUTPUT_ROOT=${OUTPUT_ROOT:-$SCRIPT_DIR}
+OUTPUT_DIR=$OUTPUT_ROOT/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$OUTPUT_ROOT/workdir/$OUTPUT_FOLDER_NAME
+max_processes=${MAX_PROCESSES:=$(python3 -c "import os; print(os.cpu_count())")}
+GCP_INGEST_SERVICE_KEY=${GCP_INGEST_SERVICE_KEY:-$GCP_INGEST_SERVICE_KEY}
+
+# shellcheck disable=SC1091
+source "$SCRIPT_DIR"/cleanup.sh
+function cleanup() {
+  cleanup_dir "$OUTPUT_DIR"
+  cleanup_dir "$WORK_DIR"
```

Comment:
Did you want to clean up the .json that gets made from authentication? Probably not a total necessity.

Reply:
Since file creation is on the embedder level, cleanup also needs to be on embedder level, or else not all cleanup cases are covered. Hence: ~~[add credentials cleanup](https://github.com/Unstructured-IO/unstructured/pull/2693/commits/78c69a1ec55f186ffd7786e954c28c8fc6f38b50)~~

At the same time, embedder instances do not know when pipelines do not need any more "embed" operations. This means, with the approach in the above strikethroughed commit, before every "embed" operation, a credentials file needs to be created and after every "embed" operation, that file needs to be deleted. This introduces repeated create/delete/create/delete... overhead.

Instead, saving the credentials to /tmp for system managed cleanups: [save creds to tmp rather than manual cleanup](https://github.com/Unstructured-IO/unstructured/pull/2693/commits/2866f42d71d02c27c1e81abb0f77122e52ff517c)

Reply:
Further explanation:

> At the same time, embedder instances do not know when pipelines do not need any more "embed" operations.

(This means the embedder does not know / does not have access to when exactly to delete the credentials file, meaning that it has to delete the credentials file every time after it makes an "embed" operation.)

Reply:
makes sense. I'm just talking for this test only. But again I think it just disposes of the container. Temp directory also totally works for this case.

I see you did the switch to temp directory in the main vertex.py file. Sounds like a good solution. Any downsides to this? Should work fine for parallel workers since they will all setup their own environments. At some point the environment variable will point to a path that is no longer there. But that shouldn't matter.

Reply:
Oh. For some reason I thought you would use python's tempfile which I think deletes the file after python closes down.
https://docs.python.org/3/library/tempfile.html
I'm sure `/tmp` works just as well